hr_employee_document/models/hr_employee.py

```diff
@@ -7,30 +7,23 @@
 class HrEmployee(models.Model):
     _inherit = "hr.employee"
 
-    document_ids = fields.One2many(
-        "ir.attachment",
-        compute="_compute_document_ids",
-        groups="hr.group_hr_user",
-        string="Documents",
-    )
     documents_count = fields.Integer(
-        compute="_compute_document_ids",
+        compute="_compute_documents_count",
         groups="hr.group_hr_user",
         string="Document Count",
     )
 
-    def _compute_document_ids(self):
-        attachments = self.env["ir.attachment"].search(
-            [("res_model", "=", self._name), ("res_id", "in", self.ids)]
+    def _compute_documents_count(self):
+        self.documents_count = 0
+        employee_obj = self.env["hr.employee"]
+        attachment_groups = self.env["ir.attachment"].read_group(
+            [("res_model", "=", self._name), ("res_id", "in", self.ids)],
+            ["res_id"],
+            ["res_id"],
         )
-
-        result = dict.fromkeys(self.ids, self.env["ir.attachment"])
-        for attachment in attachments:
-            result[attachment.res_id] |= attachment
-
-        for employee in self:
-            employee.document_ids = result[employee.id]
-            employee.documents_count = len(employee.document_ids)
+        for attachment_group in attachment_groups:
+            record = employee_obj.browse(attachment_group["res_id"])
```

Comment:
Don't rebrowse, as this is breaking caches. Just create a dictionary with current groups:
```python
count_dict = {x["res_id"]: x["res_id_count"] of x in attachment_groups}
for record in self:
    record.documents_count = count_dict.get(record.id, 0)
```